llpc/lower/llpcSpirvLowerTerminator.h

```diff
@@ -0,0 +1,61 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcSpirvLowerTerminator.h
+ * @brief LLPC header file: contains declaration of class Llpc::SpirvLowerTerminator.
+ ***********************************************************************************************************************
+ */
+#pragma once
+
+#include "SPIRVInternal.h"
+#include "llpcSpirvLower.h"
+#include "llvm/IR/InstVisitor.h"
+#include <stack>
+#include <unordered_set>
+
+namespace Llpc {
+
+// =====================================================================================================================
+// Represents the pass of SPIR-V lowering opertions for kills.
+class SpirvLowerTerminator : public SpirvLower, public llvm::InstVisitor<SpirvLowerTerminator> {
+public:
+  SpirvLowerTerminator();
+
+  virtual bool runOnModule(llvm::Module &module);
+  virtual void visitCallInst(llvm::CallInst &callInst);
+
+  static char ID; // ID of this pass
+
+private:
+  SpirvLowerTerminator(const SpirvLowerTerminator &) = delete;
+  SpirvLowerTerminator &operator=(const SpirvLowerTerminator &) = delete;
+
+  bool m_changed;
+  std::unordered_set<llvm::Instruction *> m_instsForRemoval; // Instructions to be removed
```

Comment:
`DenseSet<llvm::Instruction*>`?

Reply:
Use SmallPtrSet as the number of instructions is going to be small (typically zero).